### ISSUE TITLE
Rule Change: Let camelcase distinguish between camelCase and PascalCase

### ISSUE THREAD

**Issue by dwaard** [eslint/eslint#15530]
### What rule do you want to change?

camelcase

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
/*eslint camelcase: ["error", { allowPascalCase: false }*/
var camelCasedVariable = 'hello world'; // no error
var PascalCasedVariable = 'hello world'; // error
var Test = 0; // error

/*eslint camelcase: ["error", { allowPascalCase: true }*/
var camelCasedVariable = 'hello world'; // no error
var PascalCasedVariable = 'hello world'; // no error
var Test = 0; // no error
```

### What does the rule currently do for this code?

Variables that start with an uppercase character are formally not camelcase but pascalcase. It might be useful in some cases to allow or not allow variables to start with an upper case character.

### What will the rule do after it's changed?

It adds an option `allowPascalCase`. When set to `true` it allows pascalcase variables. If its set to `false` it will generate error

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
